providers/providers.go

```diff
@@ -103,16 +110,27 @@ func (pm *ProviderManager) getProvSet(k []byte) (*providerSet, error) {
 	}
 
 	if len(pset.providers) > 0 {
-		pm.providers.Add(string(k), pset)
+		pm.cache.Add(string(k), pset)
 	}
 
 	return pset, nil
 }
 
+// Load local providers into cache
 func loadProvSet(dstore ds.Datastore, k []byte) (*providerSet, error) {
-	res, err := dstore.Query(dsq.Query{Prefix: mkProvKey(k)})
-	if err != nil {
-		return nil, err
+	var res dsq.Results
+	if k != nil {
+		r, err := dstore.Query(dsq.Query{Prefix: mkProvKey(k)})
+		if err != nil {
+			return nil, err
+		}
+		res = r
+	} else { // Return all providers
+		r, err := dstore.Query(dsq.Query{Prefix: providersKeyPrefix})
```

Comment:
This could be..._a lot_ of data. Should we expose a separate function to stream this instead?